components/org.wso2.carbon.identity.keyrotation/src/main/java/org.wso2.carbon.identity.keyrotation/util/ConfigFileUtil.java

```diff
@@ -0,0 +1,134 @@
+
+        String[] paths = new String[]{KeyRotationConstants.REPOSITORY, KeyRotationConstants.TENANTS};
+        File[] listOfFolders = getFilePaths(isHome, paths);
+        List<String> folderPaths = new ArrayList<>();
+        for (File folder : listOfFolders) {
+            folderPaths.add(folder.getName() + FileSystems.getDefault().getSeparator());
+        }
+        return folderPaths;
+    }
+
+    /**
+     * Update the configuration file with the new re-encrypted value.
+     *
+     * @param filename          The absolute path of the configuration file.
+     * @param keyRotationConfig Configuration data needed to perform the task.
+     * @param property          The property value to identify the corresponding config file.
+     * @throws KeyRotationException Exception thrown if something unexpected happens during key rotation.
+     */
+    public static void updateConfigFile(File filename, KeyRotationConfig keyRotationConfig, String property)
+            throws KeyRotationException {
+
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+        try {
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            Document document = documentBuilder.parse(filename);
+            XPath xpath = XPathFactory.newInstance().newXPath();
+            NodeList data =
+                    (NodeList) xpath.compile("//" + property + "[@encrypted='true'][@name='password']/text()")
+                            .evaluate(document, XPathConstants.NODESET);
+            if (data.getLength() != 1 && data.getLength() != 0) {
+                throw new KeyRotationException("Error occurred while updating config file having multiple encrypted " +
+                        "properties");
+            }
+            if (data.getLength() == 1) {
+                log.info("Re-encryption in " + filename + " configuration file.");
+                String encryptedValue = data.item(0).getNodeValue();
+                log.info("Encrypted value " + encryptedValue);
+                String reEncryptedValue = reEncryptor(encryptedValue, keyRotationConfig);
+                log.info("Re-encrypted value " + reEncryptedValue);
+                data.item(0).setNodeValue(reEncryptedValue);
+                Transformer transformer = TransformerFactory.newInstance().newTransformer();
+                transformer.transform(new DOMSource(document), new StreamResult(filename.getPath()));
+            }
+        } catch (SAXException | IOException e) {
+            throw new KeyRotationException("Error occurred while parsing the xml file, " + e);
+        } catch (TransformerException e) {
+            throw new KeyRotationException("Error occurred while updating configuration file, " + e);
+        } catch (ParserConfigurationException e) {
+            throw new KeyRotationException("Error occurred while creating a new instance of DocumentBuilder, " + e);
```

Comment:
this information may not be useful to the user, we can tell a generic reason on why the config file update fails.

Reply:
Addressed with https://github.com/wso2/identity-tools/pull/13/commits/c1822aa3638f426a55388883199f2d5788aa91ae